clients/drcachesim/tracer/output.cpp

```diff
@@ -1100,8 +1100,10 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
     // Switch to instruction-tracing mode by adding FILTER_ENDPOINT marker if another
     // thread triggered the switch.
     ptr_int_t mode = tracing_mode.load(std::memory_order_acquire);
-    if (get_local_mode(data) != mode) {
-        if (get_local_mode(data) == BBDUP_MODE_L0_FILTER) {
+    ptr_int_t local_mode = get_local_mode(data);
+    if (op_L0_filter_until_instrs.get_value() && local_mode != mode &&
+        mode == BBDUP_MODE_TRACE) {
+        if (local_mode == BBDUP_MODE_L0_FILTER || local_mode == BBDUP_MODE_COUNT) {
```

Comment:
Is it possible to have a test reliably hit this?  Or the timing is too hard to guarantee?  At least discuss in the PR description how you tested this.

Reply:
I have added a test cmd line to the description. I tried adding the same test to the suite, but weirdly it produces no traces (and the test fails with `raw2trace failed: No thread files`). But when I copy paste the test cmd line (and replace the semi-colons with spaces) and run it on the terminal, it runs as expected.